src/Editor/EditorSupport.php

```diff
@@ -86,11 +66,19 @@ function () {
 	}
 
 	/**
-	 * Returns whether the editor in the current environment supports plugin features.
+	 * Returns whether the current environment is supported by the plugin
+	 * and the current screen is using the block editor.
 	 *
 	 * @return bool
 	 */
-	public function editor_supports_amp_block_editor_features() {
-		return $this->dependency_support->has_support();
+	public function is_current_screen_supported_block_editor_for_amp_enabled_post_type() {
+		$screen = get_current_screen();
+		return $this->dependency_support->has_support()
+			&&
+			$screen
+			&&
+			! empty( $screen->is_block_editor )
+			&&
+			in_array( get_post_type(), AMP_Post_Type_Support::get_supported_post_types(), true );
 	}
```

Comment:
Per above, this could instead be:

```php
	public function is_block_editor_screen_for_amp_enabled_post_type() {
		$screen = get_current_screen();
		return (
			$screen
			&&
			! empty( $screen->is_block_editor )
			&&
			in_array( get_post_type(), AMP_Post_Type_Support::get_supported_post_types(), true )
		);
	}
```